### DISCUSSION TITLE
I've tried writing data larger than ram but experience high load.

### DISCUSSION THREAD

**Discussion by hiqsociety** [erthink/libmdbx#345] [Q&A] [answered]
I've tried writing data larger than ram but experience high load.

Question
1. can database be larger than ram size?
2. what does pagesize mean and how is it used in geometry?
3. seems if ram is limited, and it's a write intensive workload, WAL is "recommended" but how often should the txn be complete? what's a good measurement of how long a transaction should run?

because if it's write intensive, would like to know how often to complete a txn so that the ram is not used up so consistently.

**Comment by erthink** [comment, marked as answer]
1. Yes. But you should understanding how b+tree-based DB work and understand the consequences of DATA > RAM (i.e. https://en.wikipedia.org/wiki/Memory_paging).
2. The size of [B-tree](https://en.wikipedia.org/wiki/B-tree) page, exactly.
The page size (obviously) can't be changed after DB was created..
So, to use custom (i.e. non-default) page size you should call [`mdbx_env_set_geometry()`](https://libmdbx.dqdkfa.ru/group__c__settings.html#ga79065e4f3c5fb2ad37a52b59224d583e) after `mdbx_env_create()`, but before `mdbx_env_open()` (which will create a new DB if it not exist).
3. There is no simple answer here:
   - Firstly, (obviously) you should commit transaction each time to made changes durable and made it visible for subsequent read transactions. This may be a principal restriction.
   - Secondary, you should rely on the number of dirty and retired pages of the transaction, i.e. use [`mdbx_txn_info()`](https://libmdbx.dqdkfa.ru/group__c__statinfo.html#ga6ec787bbb13da2028fb9c70993ef6ee1) to weigh transactions and commit when certain thresholds are exceeded.
